analysis/magefire/src/modules/features/Abilities.tsx

```diff
@@ -66,8 +66,26 @@ class Abilities extends CoreAbilities {
         category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
         gcd: null,
         cooldown: (haste: any) =>
-          (combatant.hasTalent(SPELLS.FLAME_ON_TALENT.id) ? 10 : 12) / (1 + haste),
-        charges: combatant.hasTalent(SPELLS.FLAME_ON_TALENT.id) ? 3 : 2,
+          combatant.has4Piece() && combatant.hasBuff(SPELLS.COMBUSTION.id)
+            ? 12 / 1.5 / (1 + haste)
+            : 12 / (1 + haste),
+        charges: 2,
+        enabled: !combatant.hasTalent(SPELLS.FLAME_ON_TALENT.id),
+        castEfficiency: {
+          suggestion: true,
+          recommendedEfficiency: 0.9,
+        },
+      },
+      {
+        spell: SPELLS.FIRE_BLAST.id,
+        category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        gcd: null,
+        cooldown: (haste: any) =>
+          combatant.has4Piece() && combatant.hasBuff(SPELLS.COMBUSTION.id)
+            ? 10 / 1.5 / (1 + haste)
+            : 10 / (1 + haste),
+        charges: 3,
+        enabled: combatant.hasTalent(SPELLS.FLAME_ON_TALENT.id),
```

Comment:
not a huge fan of having the duplicate entry here, but i suppose it isn't _that_ bad since the charges and cd change and the cd is nontrivial.